src/object/obj_tx.c

```diff
@@ -2051,6 +2058,8 @@ dc_tx_restart(tse_task_t *task)
 			goto out;
 		}
 
+		D_ASSERT(!tx->tx_fixed_epoch);
```

Comment:
Should we do this after acquiring tx->tx_lock instead? There's almost no performance concern. And, D_ASSERT aborts the whole process anyway, leading to no concern about holding tx->tx_lock. It takes a bit time to convince myself that not holding the lock is safe.

Reply:
The tx_fixed_epoch is set when create the TX handle, then nobody can change it, so it is unnecessary to hold tx_lock to check it.

Reply:
Better to be safe than sorry. It would also save time when people read the code.